app/renderer/js/utils/domain-util.js

```diff
@@ -101,54 +102,80 @@ class DomainUtil {
+			// since getting server settings has already failed
+			return serverConf;
+		} else {
+			// Report error to sentry to get idea of possible certificate errors
+			// users get when adding the servers
+			logger.reportSentry(new Error(error));
+			const certErrorMessage = `Do you trust certificate from ${domain}? \n ${error}`;
+			const certErrorDetail = `The organization you're connecting to is either someone impersonating the Zulip server you entered, or the server you're trying to connect to is configured in an insecure way.
+			\nIf you have a valid certificate please add it from Settings>Organizations and try to add the organization again.
+			\nUnless you have a good reason to believe otherwise, you should not proceed.
+			\nYou can click here if you'd like to proceed with the connection.`;
+
+			const response = await dialog.showMessageBox({
+				type: 'warning',
+				buttons: ['Yes', 'No'],
+				defaultId: 1,
+				message: certErrorMessage,
+				detail: certErrorDetail
+			});
+			if (response === 0) {
+				// set ignoreCerts parameter to true in case user responds with yes
+				serverConf.ignoreCerts = true;
+				try {
+					return await this.getServerSettings(domain, serverConf.ignoreCerts);
+				} catch (err) {
+					if (error === `${domain} does not have any organizations added.\
+					\nPlease contact your server administrator.`) {
+						throw new Error(error);
 					}
-				});
+					return serverConf;
+				}
+			} else {
+				throw new Error('Untrusted certificate.');
 			}
-		});
+		}
+	}
+
+	async checkOggFile({ serverConf, domain, whitelistDomains, ignoreCerts, silent }) {
+		const oggCheck = {
+			url: domain + '/static/audio/zulip.ogg',
+			...RequestUtil.requestOptions(domain, ignoreCerts)
+		};
+		try {
+			const response = await requestPromise(oggCheck);
+			if (response.statusCode === 200) {
+				// getting serverSettings has already failed
+				return serverConf;
+			}
+		} catch (err) {
+			const error = err || '';
+			const certsError = error.toString().includes('certificate');
+			if (domain.indexOf(whitelistDomains) >= 0 || certsError) {
+				try {
+					return await this.checkCertError(domain, serverConf, error, silent);
+				} catch (err) {
+					throw new Error(err);
+				}
+			} else {
+				const invalidZulipServerError = `${domain} does not appear to be a valid Zulip server. Make sure that \
+				\n (1) you can connect to that URL in a web browser and \n (2) if you need a proxy to connect to the Internet, that you've configured your proxy in the Network settings \n (3) its a zulip server \
+				\n (4) the server has a valid certificate, you can add custom certificates in Settings>Organizations`;
+				return invalidZulipServerError;
+			}
+		}
 	}
 
 	// ignoreCerts parameter helps in fetching server icon and
 	// other server details when user chooses to ignore certificate warnings
-	checkDomain(domain, ignoreCerts = false, silent = false) {
+	async checkDomain(domain, ignoreCerts = false, silent = false) {
```

Comment:
I really liked reading the change for this function, this and other changes in this commit looks great. This version if far more readable.

Reply:
Awesome, thanks! :)
Let me get a few more pressing issues out of the way and I'll begin work on converting all the code in the app to `async await`. Will let you know when I start.